### PR TITLE
Fix news generation

### PR DESCRIPTION
**Pull request by gaborbernat**
Also setup now readthedocs PR hook - see for this PR https://pytest-memray--20.org.readthedocs.build/en/20/news.html

Signed-off-by: Bernát Gábor <bgabor8@bloomberg.net>
Signed-off-by: Bernát Gábor <gaborjbernat@gmail.com>

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -97,10 +97,12 @@ line_length = 88
 known_first_party=["pytest_memray"]
 
 [tool.towncrier]
-package = "pytest-memray"
-package_dir = "src"
-filename = "NEWS.rst"
-directory = "news"
+name = "pytest-memray"
+filename = "docs/news.rst"
```
Comment (pablogsal) [inline pyproject.toml:101]: In general we prefer to have the news file at top level so it is easier to find.

File: README.md [hunk 2]
```diff
@@ -162,3 +162,17 @@ directly the make targets). You can use Docker to run the test suite on non Linu
 ```console
 docker-compose run --rm test tox
 ```
+
```
Comment (pablogsal) [inline README.md:165]: Can we move this out of the readme? I prefer to keep only user visible information here if possible

Comment (gaborbernat) [reply README.md:165]: The entire section? Where would you prefer to have it?

File: pyproject.toml [hunk 3]
```diff
@@ -97,10 +97,12 @@ line_length = 88
...
+directory = "docs/news"
+title_format = false
+issue_format = ":issue:`{issue}`"
+template = "docs/news/template.jinja2"
```
[the remark this answers is not in the document]
Comment (gaborbernat) [reply pyproject.toml:105]: The default template generates duplicate HTML anchors and also non-stable version anchors. This fixes that. See https://bloomberg.github.io/memray/changelog.html#features and https://bloomberg.github.io/memray/changelog.html#id1. Also removes the project name from the version number as the sole reason that's added I think to not generate anchors for the release too. The memray prefix here https://bloomberg.github.io/memray/changelog.html#memray-1-0-3-2022-04-21 I find redundant.

File: pyproject.toml (hunk 1 above)
Comment (pablogsal) [reply pyproject.toml:101]: What do you mean? The .rst is the final rendered document.

File: README.md (hunk 2 above)
Comment (pablogsal) [reply README.md:165]: Another file called MAINTAINANCE.md or something like that, but not in the readme, please

File: pyproject.toml (hunk 1 above)
Comment (gaborbernat) [reply pyproject.toml:101]: Not true, the rendered document is the HTML page, e.g. https://pytest-memray--20.org.readthedocs.build/en/20/news.html#

**Review by pablogsal — APPROVED** [review]

**Pull request merged**